File: delivery-info-service/src/main/java/com/potatorider/publihser/DeliveryPublisher.java
package com.potatorider.publihser;

import com.potatorider.domain.Delivery;

import reactor.core.publisher.Mono;

public interface DeliveryPublisher {

    Mono<Delivery> sendAddDeliveryEvent(Delivery delivery);

    Mono<Delivery> sendSetRiderEvent(Delivery delivery);
}

File: delivery-info-service/src/main/java/com/potatorider/publihser/DeliveryPublisher.kt
package com.potatorider.publihser

import com.potatorider.domain.Delivery
import reactor.core.publisher.Mono

interface DeliveryPublisher {
    fun sendAddDeliveryEvent(delivery: Delivery): Mono<Delivery>

    fun sendSetRiderEvent(delivery: Delivery): Mono<Delivery>
}
